File: week2/readability/readability.c
// This program calculates grade of text
// Bug: used 100 instead of 100.0; L and S didn't register as floats

#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string text = get_string("Text: ");
    int letters = 0;
    int words = 1;
    int sentences = 0;
    int length = strlen(text);

    // Count letters, words, and sentences
    for (int i = 0; i < length; i++)
    {
        char temp = text[i];
        if (isalpha(temp))
        {
            letters++;
        }
        if (isspace(temp))
        {
            words++;
        }
        if (temp == '.' || temp == '!' || temp == '?')
        {
            sentences++;
        }
    }

    // Average letters per 100 words in text
    float L = 100.0 * letters / words;

    // Average number of sentences per 100 words in text
    float S = 100.0 * sentences / words;

    // Implement Coleman-Liau index
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Print answers
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 15)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: week6/pset6/sentimental-readability/readability.py
# Readability!

text = input("Text: ")

letter_num = 0
word_num = 1
sentence_num = 0

for char in text:
    if char.isalpha():
        letter_num += 1
    if char.isspace():
        word_num += 1
    if char == "!" or char == "?" or char == ".":
        sentence_num += 1

L = 100 * letter_num / word_num
S = 100 * sentence_num / word_num
index = round(0.0588 * L - 0.296 * S - 15.8)

if index < 1:
    print("Before Grade 1")
elif index > 15:
    print("Grade 16+")
else:
    print(f"Grade {index}")
